src/pages/reportPropTypes.js

```diff
@@ -0,0 +1,76 @@
+import _ from 'underscore';
+import PropTypes from 'prop-types';
+import CONST from '../CONST';
+
+export default PropTypes.shape({
+
+    /** The specific type of chat */
+    chatType: PropTypes.oneOf(['', ..._.values(CONST.REPORT.CHAT_TYPE)]),
+
+    /** Whether there is an outstanding amount in IOU */
+    hasOutstandingIOU: PropTypes.bool,
+
+    /** List of icons for report participants */
+    icons: PropTypes.arrayOf(PropTypes.string),
+
+    /** Are we loading more report actions? */
+    isLoadingMoreReportActions: PropTypes.bool,
+
+    /** Flag to check if the report actions data are loading */
+    isLoadingReportActions: PropTypes.bool,
+
+    /** Whether the user is not an admin of policyExpenseChat chat */
+    isOwnPolicyExpenseChat: PropTypes.bool,
+
+    /** Indicates if the report is pinned to the LHN or not */
+    isPinned: PropTypes.bool,
+
+    /** The email of the last message's actor */
+    lastActorEmail: PropTypes.string,
+
+    /** The text of the last message on the report */
+    lastMessageText: PropTypes.string,
+
+    /** The time of the last message on the report */
+    lastMessageTimestamp: PropTypes.number,
+
+    /** The sequence number of the last action read by the user */
+    lastReadSequenceNumber: PropTypes.number,
+
+    /** The last time the report was visited */
+    lastVisitedTimestamp: PropTypes.number,
+
+    /** The largest sequenceNumber on this report */
+    maxSequenceNumber: PropTypes.number,
+
+    /** The current user's notification preference for this report */
+    notificationPreference: PropTypes.oneOfType([
+        // Some old reports have numbers for the notification preference
+        PropTypes.number,
+        PropTypes.string,
+    ]),
+
+    /** The policy name to use for an archived report */
+    oldPolicyName: PropTypes.string,
+
+    /** The email address of the report owner */
+    ownerEmail: PropTypes.string,
+
+    /** List of primarylogins of participants of the report */
+    participants: PropTypes.arrayOf(PropTypes.string),
+
+    /** Linked policy's ID */
+    policyID: PropTypes.string,
+
+    /** Name of the report */
+    reportName: PropTypes.string,
+
+    /** ID of the report */
+    reportID: PropTypes.string.isRequired,
```

Comment:
Question: since reportID is required here, won't it break the usages where reportID was not required (or even existed) in the props where it was not necessary? files?

Reply:
Yeah, so I ended up removing the `.isRequired` here, however, I not sure what you mean by "files". A report cannot exist in Onyx without a reportID since the collection is keyed by reportID (but not all reports going to the props comes straight from Onyx either).

Reply:
I mean other JS files like [src/pages/home/report/ReportActionsList.js](https://github.com/Expensify/App/pull/11362/files#diff-9ac76d645ad53156c86195c1a8f5adb3bb9b71f78351e43c4e1b6d6896921c4e), that was not requiring reportID. 

But I think removing solves it! Thanks!